tools/crm_resource_runtime.c

```diff
@@ -1710,6 +1710,14 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
         free(level);
     }
 
+    if (pcmk__str_eq(out->fmt_name, "xml", pcmk__str_none)) {
+        resource_verbose = 1;
+
+        if (pcmk__str_eq(action, "validate-all", pcmk__str_none)) {
+            pcmk__output_xml_create_parent(out, "validate", NULL);
+        }
```

Comment:
I'm confused by the schema changes. Aren't we already creating a new result element in the previous commit? Why would we also need this? Shouldn't the schema change have the new result element?

Reply:
The resulting XML looks like this:
```
# crm_resource -r dummy --validate --output-as xml 
<pacemaker-result api-version="2.3" request="crm_resource -r dummy --validate --output-as xml">
  <validate>
    <overrides/>
    <result status="0" action="validate-all" class="ocf" type="Dummy" rsc="dummy" provider="heartbeat"/>
    <command code="0">
      <output source="stderr">DEBUG: dummy validate-all : 0
</output>
    </command>
  </validate>
  <status code="0" message="OK"/>
</pacemaker-result>
```

Reply:
What will it look like for a force-\* command? Presumably just the \<overrides\> and \<result\>? I think having a common element instead of validate would be better. Currently, force-\* has only the status for XML:

    <pacemaker-result api-version="2.3" request="crm_resource -r rsc1 --force-start --output-as=xml">
      <status code="0" message="OK"/>
    </pacemaker-result>

Looking at the new output, I think \<result\> is confusing from a semantic standpoint. I'm thinking along these lines:

    <pacemaker-result api-version="2.3" request="crm_resource -r rsc1 --force-start --output-as=xml">
      <resource-agent-action rsc="rsc1" class="ocf" provider="heartbeat" type="Dummy" action="start">
         <overrides/>
         <agent-status code="0" message="ok"/> <!-- services_ocf_exitcode_str() -->
         <!-- any output here -->
      </resource-agent-action>
      <status code="0" message="OK"/>
    </pacemaker-result>

BTW, aren't we supposed to bump PCMK__API_VERSION manually? We should do that in 2.1 as well ...

Reply:
```
# crm_resource -r dummy --force-stop --output-as xml
<pacemaker-result api-version="2.3" request="crm_resource -r dummy --force-stop --output-as xml">
  <overrides/>
  <result status="0" action="stop" class="ocf" type="Dummy" rsc="dummy" provider="heartbeat"/>
  <status code="0" message="OK"/>
</pacemaker-result>
```

I like the resource-agent-action tag with all its attributes more than what I originally had.  Also, looking at it in the --force-stop case, result and status right next to each other is confusing.

Yes, we were almost certainly supposed to have bumped PCMK__API_VERSION at some point.  Ugh, too many version numbers to keep track of.

Reply:
Updated XML output and increased PCMK__API_VERSION.